File: Day6-TodoList/app/src/main/java/com/byted/camp/todolist/ViewController/Adapters/NoteListAdapter.java
package com.byted.camp.todolist.ui;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


import com.byted.camp.todolist.R;
import com.byted.camp.todolist.TodoListViewModel;
import com.byted.camp.todolist.db.room.TodoEntity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created on 2019/1/23.
 *
 * @author xuyingyi@bytedance.com (Yingyi Xu)
 */
public class NoteListAdapter extends RecyclerView.Adapter<NoteViewHolder> {

    private final TodoListViewModel viewModel;
    private final List<TodoEntity> notes = new ArrayList<>();

    public NoteListAdapter(TodoListViewModel viewModel) {
        this.viewModel = viewModel;
    }

    public void refresh(List<TodoEntity> newNotes) {
        notes.clear();
        if (newNotes != null) {
            notes.addAll(newNotes);
        }
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public NoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int pos) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_note, parent, false);
        return new NoteViewHolder(itemView, viewModel);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteViewHolder holder, int pos) {
        holder.bind(notes.get(pos));
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }
}

File: Day6-TodoList/app/src/main/java/com/byted/camp/todolist/viewController/adapters/NoteListAdapter.kt
package com.byted.camp.todolist.viewController.adapters

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup


import com.byted.camp.todolist.R
import com.byted.camp.todolist.viewModel.TodoListViewModel
import com.byted.camp.todolist.db.room.TodoEntity

import java.util.ArrayList

/**
 * Created on 2019/1/23.
 *
 * @author xuyingyi@bytedance.com (Yingyi Xu)
 */
class NoteListAdapter(private val viewModel: TodoListViewModel) : RecyclerView.Adapter<NoteViewHolder>() {
    private val notes = ArrayList<TodoEntity>()

    fun refresh(newNotes: List<TodoEntity>?) {
        notes.clear()
        if (newNotes != null) {
            notes.addAll(newNotes)
        }
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, pos: Int): NoteViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_note, parent, false)
        return NoteViewHolder(itemView, viewModel)
    }

    override fun onBindViewHolder(holder: NoteViewHolder, pos: Int) {
        holder.bind(notes[pos])
    }

    override fun getItemCount(): Int {
        return notes.size
    }
}
